plugin/virustotal/vtreport.py

```diff
@@ -0,0 +1,404 @@
+# Copyright 2020 Google Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__author__ = 'gerardofn@virustotal.com'
+
+from datetime import datetime
+import logging
+import vt
+
+SUPPORTED_FILES = ('PEEXE', 'PEDLL', 'MZ', 'MSI', 'COM',
+                   'COFF', 'ELF', 'LINUX', 'MACHO')
+
+
+class VTReport(object):
+
+  def __init__(self, apikey, file_hash, privileges):
+    global SUPPORTED_FILES
+    client = vt.Client(apikey)
+    general = client.get_object('/files/{}', file_hash)
+    file_type = general.get('type_tag')
+    self.private_api = privileges
+    self.compressed_parents = []  # List of dictionaries
+    self.dropped_files = []       # List of dictionaries
+    self.execution_parents = []   # List of dictionaries
+    self.contacted_ips = {}
+    self.contacted_urls = {}
+    self.contacted_domains = {}
+    self.embedded_domains = {}
+    self.embedded_ips = {}
+    self.embedded_urls = {}
+    self.yara_results = []        # List of dictionaries
+    self.comments = {}  
+    self.behaviour = {}
+    self.filenames = []           # List of strings
+    self.sigma_results = {}
+
+    if file_type.upper() in SUPPORTED_FILES:
+      logging.debug('[VT Report] Processing file type %s.', file_type.upper())
+
+      # Public API Key
+      dets = general.get('last_analysis_stats')
+      total = dets['harmless'] + dets['malicious']
+      total += dets['suspicious'] + dets['undetected']
+      self.detections_malicious = str(dets['malicious'])
+      self.detections_total = str(total)
+
+      if file_type in ('PEEXE', 'PEDLL'):
+        cdate = general.get('creation_date')
+        self.creation = datetime.utcfromtimestamp(cdate)
+        self.creation = self.creation.strftime('%Y-%m-%d %H:%M:%S')
+      else:
+        self.creation = None
+
+      fsdate = general.get('first_submission_date')
```

Comment:
This is creating a variable in a very long function, when it is just being used once a couple lines afterwards. Please instead avoid creating the variable and put the expression directly in its use -- this helps prevent bugs later on when modifying code and reusing the same variable in a un-expected way.

( do the same for any other variables )